p2pserver/protocols/discovery/discovery.go

```diff
@@ -34,24 +34,26 @@ import (
 )
 
 type Discovery struct {
-	dht     *dht.DHT
-	net     p2p.P2P
-	id      common.PeerId
-	quit    chan bool
-	maskSet *strset.Set
+	dht        *dht.DHT
+	net        p2p.P2P
+	id         common.PeerId
+	quit       chan bool
+	maskSet    *strset.Set
+	maskFilter p2p.AddressFilter //todo : conbine with maskSet
```

Comment:
I do not think so. they are all used for filtering some addresses. there is no difference for discovery.  so only need one `AddressFilter` field, and set with `CombinedAddressFilter(maskSetFilter, subnetMaskFilter)`

Reply:
maskset works with reserve-list to provide sub-net functions before.

reserve-list defines sub-net, maskset serves as gateway, although they may have same ip list.
subnet is more like reserve-list, rather than maskset.

Reply:
discussed in wechat. subnet should act like reserve-list and maskset to hide gov node.